src/hotspot/share/opto/subnode.hpp

```diff
@@ -297,6 +297,89 @@ class CmpD3Node : public CmpDNode {
   virtual uint ideal_reg() const { return Op_RegI; }
 };
 
+// Does a ScopedValue.get() hits in the cache?
```

Comment:
I finally reconstructed what this node is, and please add a comment saying something like this:
This node returns true iff this gets us a cache hit (cache reference not null, and at least one of the indices leads to a hit).
It is essencially a Cmp, comparing the `cache_adr` (you name it scoped_value_cache) with a nullptr.
But it also gets 2 indices to that cache (ints), which will either score a hit or miss.

Reply:
Thanks for the update :heavy_check_mark: